backend/papersfeed/utils/recommendations/utils.py

```diff
@@ -0,0 +1,60 @@
+"""utils.py"""
+# -*- coding: utf-8 -*-
+from datetime import datetime, timedelta
+from django.db.models import OuterRef, Subquery, F
+
+from papersfeed.models.papers.paper import Paper
+from papersfeed.models.users.user import User
+from papersfeed.models.users.user_action import UserAction
+from papersfeed.models.users.user_recommendation import UserRecommendation
+
+def select_user_actions(_):
+    """Select user actions"""
+    new_actions = UserAction.objects.filter(
+        modification_date__gt=(datetime.now() + timedelta(days=-1))
+    ).annotate(
+        UserId=__get_user_id('user'),
+        ItemId=__get_paper_id('paper'),
+        Type=F('type'),
+        Count=F('count'),
+    ).values(
+        'UserId', 'ItemId', 'Type', 'Count'
+    )
+
+    new_actions = list(new_actions)
+
+    for action in UserAction.objects.all():
+        setattr(action, 'count', 0)
+        action.save()
+
+    return new_actions
+
+def insert_user_recommendation(args):
+    """Insert user recommendation"""
+
+    for data in args['data']:
+        user_id = data['user']
+        papers = data['papers']
+
+        for i, paper_id in enumerate(papers):
```

Comment:
I think there may be more Django-tic way, also.

Reply:
I should think about this problem more ㅠㅠ

Reply:
If you have some troubles about it, I think we can postpone this issue for refactoring.

Reply:
Yes I should doㅠ Thank you